stan/math/prim/meta/vector_seq_view.hpp

```diff
@@ -79,43 +52,17 @@ class vector_seq_view<Eigen::Matrix<S, 1, Eigen::Dynamic> > {
  * only allows std::vectors as the container type, since we would have
  * difficulty figuring out which contained type was the container otherwise.
  *
- * @tparam S the type inside of the underlying Vector
+ * @tparam S the type inside of the std::vector
  */
-template <typename S>
-class vector_seq_view<std::vector<Eigen::Matrix<S, Eigen::Dynamic, 1> > > {
- public:
-  explicit vector_seq_view(
-      const std::vector<Eigen::Matrix<S, Eigen::Dynamic, 1> >& v)
-      : v_(v) {}
-  int size() const { return v_.size(); }
-  Eigen::Matrix<S, Eigen::Dynamic, 1> operator[](int i) const { return v_[i]; }
-
- private:
-  const std::vector<Eigen::Matrix<S, Eigen::Dynamic, 1> >& v_;
-};
-
-/** \ingroup type_trait
- * This class provides a low-cost wrapper for situations where you either need
- * an Eigen Vector or RowVector or a std::vector of them and you want to be
- * agnostic between those two options. This is similar to scalar_seq_view but
- * instead of being a sequence-like view over a scalar or seq of scalars, it's
- * a sequence-like view over a Vector or seq of Vectors. Notably this version
- * only allows std::vectors as the container type, since we would have
- * difficulty figuring out which contained type was the container otherwise.
- *
- * @tparam S the type inside of the underlying RowVector
- */
-template <typename S>
-class vector_seq_view<std::vector<Eigen::Matrix<S, 1, Eigen::Dynamic> > > {
+template <typename T>
+class vector_seq_view<T, require_std_vector_vt<is_container, T>> {
  public:
-  explicit vector_seq_view(
-      const std::vector<Eigen::Matrix<S, 1, Eigen::Dynamic> >& v)
-      : v_(v) {}
+  explicit vector_seq_view(const T& v) : v_(v) {}
   int size() const { return v_.size(); }
-  Eigen::Matrix<S, 1, Eigen::Dynamic> operator[](int i) const { return v_[i]; }
+  const value_type_t<T>& operator[](int i) const { return v_[i]; }
```

Comment:
If the input is a `std::vector` of expressions, won't it re-evaluate them each time it returns here? Or is `std::vector` of expressions not something that'll ever be tossed around in Stan?

Reply:
Yeah, as I started working on those I never even thought of `std::vector` of expressions being possible. I still don't think it would be an often used type, so I keep pretending it does not exist. So for now we will not have functions return such types. We can always add support for it later.